storage/jellyfish-merkle/src/lib.rs

```diff
@@ -655,3 +656,78 @@ where
             .map(|root_node| root_node.hash()))
     }
 }
+
+trait NibbleExt {
```

Comment:
So NibbleExt is only used in Merkle trees?

Reply:
Yeah these nibble functions on the `HashValue` type are only ever used in this module (and I think they were added explicitly for the merkle trees)

Reply:
I applaude the cleaning